### PR TITLE
Fix for PARKING_EXTRUDER homing and solenoid control

### PR DESCRIPTION
**Pull request by zeleps**
### Description

PARKING_EXTRUDER feature is -apparently- not widely used and it appears that, through the many changes in the homing operation, the feature broke. As it is evident from the code remnants, the intention was to finish homing with the correct extruder unparked. This does not happen, and also there never was a provision for supporting an X_MAX homing configuration. Also, M380/381 gcodes did not take PARKING_EXTRUDER solenoid configuration into account, thus working in reverse under certain conditions. The proposed solution minimizes the movements necessary to unpark the right extruder after homing, while considering if the homing operation begins with no extruder attached and respecting the differences in homing direction. 

### Benefits

PARKING_EXTRUDER homing and solenoid control now work as initially intended

### Configurations

Only PARKING_EXTRUDER needs to be enabled

### Related Issues

None, it seems that this feature is not widely used, but it was broken nonetheless, so a fix was needed.

### Integration Testing

1. Enable and configure PARING_EXTRUDER option.

2. Perform the following actions and verify results:

- G28: Carriage homes X towards E0 (active) and grabs it, then completes Z-homing.
- G28: Carriage homes with E0 attached
- T1: Carriage parks E0, grabs E1
- G28: Carriage homes with E1 attached
- M381: Carriage releases E1 (E1 should be parked manually)
- T1: Carriage directly grabs E1
- M381: Carriage releases E1 (E1 should be parked manually)
- T0: Carriage directly grabs E0
- T1: Carriage parks E0, grabs E1
- M381: Carriage releases E1 (E1 should be parked manually)
- G28: Carriage homes X towards E0, completes Z homing and then grabs E1 directly

3. Reverse X-Axis homing (can be *temporarily* done in-place by reversing solenoid pins, endstop pins, USE_XMAX_PLUG, X_HOME_DIR and INVERT_X_DIR)

4. Perform the following actions and verify results:

- G28: Carriage homes X towards E1, completes Z-homing, then grabs E0
- G28: Carriage homes with E0 attached
- T1: Carriage parks E0, grabs E1
- G28: Carriage homes with E1 attached
- M381: Carriage releases E1 (E1 should be parked manually)
- T1: Carriage directly grabs E1
- M381: Carriage releases E1 (E1 should be parked manually)
- T0: Carriage directly grabs E0
- T1: Carriage parks E0, grabs E1
- M381: Carriage releases E1 (E1 should be parked manually)
- G28: Carriage homes X towards E1 and grabs it, completes Z homing with E1 attached

### CODE REVIEW

**Comment by thinkyhead** [comment]
Thanks for taking the time to work out the issues. It's not always easy to find testers for these _interesting_ features.

**Comment by zeleps** [comment]
Cheers @thinkyhead, your appraisal means a lot! I've been using Marlin for a couple of years now, this is the first time I took the initiative to contribute, mainly due to building a new printer from the scratch as a past-time project (now with the lockdown and all). I'll try to do it more in the future. Keep up the good work!

File: Marlin/src/feature/solenoid.cpp [hunk 1]
```diff
@@ -32,16 +32,12 @@
 
 #if ENABLED(PARKING_EXTRUDER) 
   #include "../module/tool_change.h"
-  #define SOLENOID_MAGNETIZED_STATE TERN(PARKING_EXTRUDER_SOLENOIDS_INVERT, !PARKING_EXTRUDER_SOLENOIDS_PINS_ACTIVE, PARKING_EXTRUDER_SOLENOIDS_PINS_ACTIVE)
+  #define SOLENOID_MAGNETIZED_STATE (TERN_(PARKING_EXTRUDER_SOLENOIDS_INVERT,!)PARKING_EXTRUDER_SOLENOIDS_PINS_ACTIVE)
```
Comment (zeleps) [inline Marlin/src/feature/solenoid.cpp:35]: Neat!

**Comment by thinkyhead** [comment]
Thanks for the patch! Just had to do some final cleanup, and will merge shortly….

**Pull request merged**